pkg/stub/interfaces.go

```diff
@@ -141,13 +142,19 @@ func (g *DefaultImageStreamFromFileGetter) Get(fullFilePath string) (is *imagev1
 	if err != nil {
 		return nil, err
 	}
-	switch {
-	case imagestream.Name == "jenkins":
-		imagestream = tagInPayload("2", "IMAGE_JENKINS", imagestream)
-	case imagestream.Name == "jenkins-agent-maven":
-		imagestream = tagInPayload("v4.0", "IMAGE_AGENT_MAVEN", imagestream)
-	case imagestream.Name == "jenkins-agent-nodejs":
-		imagestream = tagInPayload("v4.0", "IMAGE_AGENT_NODEJS", imagestream)
+	if g.h != nil {
+		// if we are not skipping jenkins imagestream, tag in images from payload
```

Comment:
why does this need to be special cased?

Reply:
file processing can occur regardless of actual imagestream upserting

Reply:
I'll add a comment

Reply:
but why not just always do this?

Reply:
(ie always update the imagestream w/ the payload pullspec, even if the user is "skipping" the imagestream).

If anything it seems safer than returning an un-mutated imagestream that contains content we know is wrong.

Reply:
I think there will be cases when users want to import their own jenkins images ... at least at the beginning of 4.0 dev @jupierce relayed to me his team would need this

The agent streams yeah maybe it just makes sense to always mutate

Reply:
though per the other comment thread maybe we do allow manage separately, and allow to skip :-)

Reply:
What I was trying to get at is that I see calling this function as orthogonal to whether the imagestream is being skipped or not.

This function's job(as i understand it) is to return an imagestream for a given filepath.  (and in the case of jenkins, handle mutating the imagestream as needed, though maybe this isn't the right place to do the mutation since presumably the other imagestream mutations, such as to change the registry hostname, are done elsewhere).

It shouldn't care whether a given imagestream is in the skipped list or not, that should be up to the caller to filter out.

So making this function aware of the skiplist, specifically to handle jenkins mutation, seems awkward.

What would happen if this function always mutated the jenkins imagestream before returning it, regardless of the skiplist?

Alternatively, can we put the mutation in the same place other imagestream mutation is done?

Reply:
Admittedly the path of least resistance was here.  I'll revisit isolating the mutation, placing it elsewhere, and we can compare the awkwardness, etc.

Reply:
yeah it will be cleaner ... part of next push